File: src/main/java/org/eln/sim/mna/component/IAbstractor.java
package org.eln.sim.mna.component;

/*
NOTE: DO NOT IMPORT MINECRAFT CODE IN THIS CLASS
EXTEND IT INSTEAD IN THE org.eln.nbt DIRECTORY
 */

import org.eln.sim.mna.SubSystem;

public interface IAbstractor {

    void dirty(Component component);

    SubSystem getAbstractorSubSystem();
}

File: src/main/java/org/eln2/sim/mna/component/IAbstractor.kt
package org.eln2.sim.mna.component

/*
NOTE: DO NOT IMPORT MINECRAFT CODE IN THIS CLASS
EXTEND IT INSTEAD IN THE org.eln.nbt DIRECTORY
 */

import org.eln2.sim.mna.SubSystem

interface IAbstractor {
    fun dirty(component: Component?)
    fun getAbstractorSubSystem(): SubSystem?
}